src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -47,6 +49,40 @@ public Optional<String> getDerivedTableName(UUID entityTypeId) {
       .fetchOptional(derivedTableNameField);
   }
 
+  public List<EntityTypeColumn> fetchNamesForSingleCheckbox(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getDerivedTableName(entityTypeId).get();
+    List<EntityTypeColumn> entityColumnsList = new ArrayList<>();
+
+    String requiredFieldName = "jsonb" + "->>'name'";
+    String abc =  "jsonb" + "->>'refId'";
+    String whereClauseName = "jsonb" + "->>'type'";
+    String customFieldType = "SINGLE_CHECKBOX";
+    ValueWithLabel valueWithLabelTrue = new ValueWithLabel().label("True").value("true");
+    ValueWithLabel valueWithLabelFalse = new ValueWithLabel().label("False").value("false");
+
+    Result<Record2<Object, Object>> result = jooqContext
+      .select(field(requiredFieldName), field(abc))
+      .from(sourceViewName)
+      .where(field(whereClauseName).eq(customFieldType))
+      .fetch();
+    result.stream()
+      .map(record -> {
+        Object value = record.get(0);
+        EntityTypeColumn entityTypeColumn = new EntityTypeColumn();
+        entityTypeColumn.name((value instanceof List) ?
```

Comment:
Is this ever a `List`? If I'm understanding the code, it's this just the name of the custom field?

Reply:
@mweaver-ebsco  yes you are right it just a name not a list. Yesterday there was some thing With the name so it wasn't working properly. No Its all good. I guess we don't need to write it as list and all that code.
We can just have `entityTypeColumn.name(value.toString());`